src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -188,15 +190,33 @@
  * protected) or that there are get and set methods that can be used to restore
  * the state.
  *
- * <p>The contents of the stream can be filtered during deserialization.
- * If a {@linkplain #setObjectInputFilter(ObjectInputFilter) filter is set}
- * on an ObjectInputStream, the {@link ObjectInputFilter} can check that
+ * The deserialization filter for a stream is determined in one of the following ways:
```

Comment:
Should this line start with a <p> ?

Reply:
At this point in the OIS spec it would be good to introduce the idea of a stream-specific filter. Maybe open this paragraph with some additional context, like say:

```
 * <p> An {@code ObjectInputStream} has an optional <i>stream-specific</i>
 * deserialization filter. The stream-specific deserialization filter
 * is determined in one of the following ways:
```

then enumerate the ways in which the stream-specific deserialization filter is determine, and drop how the filter factory can be set, etc. That is best left to the Config.